crates/biome_service/src/settings.rs

```diff
@@ -30,62 +30,115 @@ use indexmap::IndexSet;
-    pub files: FilesSettings,
-    /// Analyzer settings
-    pub organize_imports: OrganizeImportsSettings,
-    /// overrides
-    pub override_settings: OverrideSettings,
+    data: WorkspaceData<PathToSettings>,
+    current_workspace: WorkspaceKey,
 }
 
 impl WorkspaceSettings {
-    /// Retrieves the settings of the formatter
-    pub fn formatter(&self) -> &FormatSettings {
-        &self.formatter
+    pub fn current_settings(&self) -> &Settings {
+        debug_assert!(
+            !self.data.paths.is_empty(),
+            "You must have at least one workspace."
+        );
+        let data = self.data.paths.get(self.current_workspace).unwrap();
+        &data.settings
     }
 
-    /// Whether the formatter is disabled for JavaScript files
-    pub fn javascript_formatter_disabled(&self) -> bool {
-        let enabled = self.languages.javascript.formatter.enabled.as_ref();
-        enabled == Some(&false)
+    pub fn register_current_workspace(&mut self, key: WorkspaceKey) {
+        self.current_workspace = key;
     }
 
-    /// Whether the formatter is disabled for JSON files
-    pub fn json_formatter_disabled(&self) -> bool {
-        let enabled = self.languages.json.formatter.enabled.as_ref();
-        enabled == Some(&false)
+    pub fn current_settings_mut(&mut self) -> &mut Settings {
+        debug_assert!(
+            !self.data.paths.is_empty(),
+            "You must have at least one workspace."
+        );
+        &mut self.data.get_mut(self.current_workspace).unwrap().settings
     }
 
-    /// Whether the formatter is disabled for CSS files
-    pub fn css_formatter_disabled(&self) -> bool {
-        let enabled = self.languages.css.formatter.enabled.as_ref();
-        enabled == Some(&false)
+    pub fn insert_workspace(&mut self, workspace_path: impl Into<PathBuf>) -> WorkspaceKey {
+        self.data.paths.insert(PathToSettings {
+            path: BiomePath::new(workspace_path.into()),
+            settings: Settings::default(),
+        })
     }
 
-    /// Retrieves the settings of the linter
-    pub fn linter(&self) -> &LinterSettings {
-        &self.linter
+    ///
+    pub fn current_workspace_path(&self, path: &BiomePath) -> (WorkspaceKey, &BiomePath) {
+        debug_assert!(path.is_absolute(), "Workspaces paths must be absolutes.");
+        debug_assert!(
+            !self.data.paths.is_empty(),
+            "You must have at least one workspace."
+        );
+        for (key, path_to_settings) in &self.data.paths {
+            if path.strip_prefix(path_to_settings.path.as_path()).is_ok() {
+                return (key, &path_to_settings.path);
+            }
+        }
+        unreachable!("We should not reach here, the assertions should help.")
     }
 
-    /// Retrieves the settings of the organize imports
-    pub fn organize_imports(&self) -> &OrganizeImportsSettings {
-        &self.organize_imports
+    pub fn get_settings_by_path(&self, path: &BiomePath) -> &Settings {
```

Comment:
So these two `get_settings_by_path` methods seem to be the only places where we iterate over the paths, and I don't the methods being used anywhere. Do we anticipate that they will be commonly used going forward?

I'm asking because the `DenseSlotMap` was chosen because of fast iteration, but at least *as of yet* we don't do any iteration yet. Based on the current state that would not make it a good choice, but that could change if these methods become the most commonly used ones in the future.

Reply:
I remove two functions and kept only one: `get_settings_by_path`.

As mentioned in the description, in the LSP there's way to retrieve the current project when handling a file. When the LSP sends us the request `textDocument/didOpen`, we only get its absolute URL. 

In order to understand if a user opened a file that belongs to a different project from the previous, we need to inspect its path, and that's where the iteration will come into play. 

If you come up with a better solution by the time I will open the next PR, that would be awesome.